crates/server/src/rpc/service.rs

```diff
@@ -23,6 +25,35 @@ type QuestRpcResult<T> = Result<T, QuestError>;
 
 #[async_trait::async_trait]
 impl QuestsServiceServer<QuestsRpcServerContext, QuestError> for QuestsServiceImplementation {
+    async fn get_quest_offer(
+        &self,
+        request: QuestOfferRequest,
+        context: ProcedureContext<QuestsRpcServerContext>,
+    ) -> QuestRpcResult<QuestOfferResponse> {
+        if let Ok(quest) = get_quest(context.server_context.db.clone(), &request.quest_id).await {
+            let user_subscriptions_lock = context.server_context.transport_contexts.read().await;
+            if let Some(transport_context) = user_subscriptions_lock.get(&context.transport_id) {
+                if let Some(subscription) = &transport_context.subscription {
+                    let user_update = UserUpdate {
+                        message: Some(Message::QuestOffer(QuestOffering {
+                            id: request.quest_id,
+                            name: quest.name,
+                            description: quest.description,
+                            definition: Some(quest.definition),
+                        })),
+                    };
+                    if subscription.r#yield(user_update).await.is_err() {
+                        log::error!("QuestServiceImplementation > StartQuest Error > Not able to send update to susbcription")
+                    }
+                }
+            }
+
+            Ok(QuestOfferResponse { offered: true })
```

Comment:
I think we should set `offered: true` just for the case when there is a subscription and could send the update. In that case, should we notify the user that they should subscribe first?

Reply:
I agree but I am not sure how can we handle the notification in order to subscribe